pages/api/projects/[id]/tasks/[taskId].js

```diff
@@ -16,5 +16,17 @@ export default async function handler(request, response) {
       return response.status(404).json({ status: "Task not found" });
     }
   }
+
+  if (request.method === "PUT") {
+    try {
+      const taskData = request.body;
+      await Task.findByIdAndUpdate(taskId, taskData);
+
+      return response.status(201).json({ message: "Task updated" });
```

Comment:
[201](https://developer.mozilla.org/en-US/docs/Web/HTTP/Status/201) is for new created resources, 200 is fine for PUT ^^

It's also good practice to send back the updated resource (even if you don't use it in the client):

```js
const updatedTask = await Task.findByIdAndUpdate(taskId, request.body, { new: true });
return response.status(200).json(updatedTask);
 ```